Lint all the things!

Per [package.json](https://github.com/mozilla/idea-town-addon/blob/2d00d202c5c742fba5995faefe8e53c18d9f2f13/package.json#L15) it looks like we're currently only linting the index.js file:
```js
"lint": "eslint index.js",
```

When I change that to lint all things (`$ eslint .`), I get some mo' errors and warnings:
```sh
➜  idea-town-addon git:(master) npm run lint

> idea-town-addon@0.0.1 lint /Users/pdehaan/dev/github/idea-town-addon
> eslint .


data/message-bridge.js
  3:25  error  Strings must use singlequote                  quotes
  3:54  error  Unexpected space before function parentheses  space-before-function-paren
  7:43  error  Unexpected space before function parentheses  space-before-function-paren
  8:3   error  Unexpected var, use let or const instead      no-var

index.js
  26:24  warning  A function with a name starting with an uppercase letter should only be used as a constructor  new-cap

stub-content/index.js
   1:1   error    Unexpected var, use let or const instead                 no-var
   4:3   error    Unexpected var, use let or const instead                 no-var
   7:3   warning  Unexpected console statement                             no-console
  10:25  error    Strings must use singlequote                             quotes
  10:54  error    Unexpected space before function parentheses             space-before-function-paren
  14:5   error    Expected indentation of 2 space characters but found 4   indent
  15:7   error    Expected indentation of 4 space characters but found 6   indent
  16:7   error    Expected indentation of 4 space characters but found 6   indent
  17:5   error    Expected indentation of 2 space characters but found 4   indent
  18:7   error    Expected indentation of 4 space characters but found 6   indent
  19:7   error    Expected indentation of 4 space characters but found 6   indent
  20:5   error    Expected indentation of 2 space characters but found 4   indent
  21:7   error    Expected indentation of 4 space characters but found 6   indent
  21:7   warning  Unexpected console statement                             no-console
  22:7   error    Expected indentation of 4 space characters but found 6   indent
  26:21  error    Unexpected space before function parentheses             space-before-function-paren
  36:3   error    Unexpected var, use let or const instead                 no-var
  37:5   error    Unexpected var, use let or const instead                 no-var
  38:5   error    Unexpected var, use let or const instead                 no-var
  39:5   error    Unexpected var, use let or const instead                 no-var
  50:3   error    Unexpected var, use let or const instead                 no-var
  58:61  error    ev is defined but never used                             no-unused-vars
  59:3   error    Unexpected var, use let or const instead                 no-var
  61:4   error    Expected indentation of 24 space characters but found 2  indent
  61:45  error    Missing semicolon                                        semi
  64:2   error    Missing semicolon                                        semi

test/test-index.js
   1:1   error  Unexpected var, use let or const instead  no-var
   1:5   error  main is defined but never used            no-unused-vars
   1:20  error  Strings must use singlequote              quotes
   3:9   error  Strings must use singlequote              quotes
   4:15  error  Strings must use singlequote              quotes
   7:9   error  Strings must use singlequote              quotes
   8:15  error  Strings must use singlequote              quotes
  12:9   error  Strings must use singlequote              quotes

✖ 39 problems (36 errors, 3 warnings)
```

Not sure if we want to lint all the files, and exclude certain files/folders using the .eslintignore file, but thought I'd float this out there.

--- a/.eslintignore
+++ b/.eslintignore
@@ -1,0 +1,2 @@
+bootstrap.js
+stub-content/index.js

--- a/data/message-bridge.js
+++ b/data/message-bridge.js
@@ -1,11 +1,11 @@
 // Page script acts as messaging bridge between addon and web content.
 
-window.addEventListener("from-web-to-addon", function (event) {
+window.addEventListener('from-web-to-addon', function(event) {
   self.port.emit('from-web-to-addon', event.detail);
 }, false);
 
-self.port.on('from-addon-to-web', function (data) {
-  var clonedDetail = cloneInto(data, document.defaultView);
+self.port.on('from-addon-to-web', function(data) {
+  const clonedDetail = cloneInto(data, document.defaultView);
   document.documentElement.dispatchEvent(new CustomEvent(
     'from-addon-to-web', { bubbles: true, detail: clonedDetail }
   ));

--- a/package.json
+++ b/package.json
@@ -12,7 +12,7 @@
   },
   "scripts": {
     "start": "st -nc .",
-    "lint": "eslint index.js",
+    "lint": "eslint .",
     "package": "jpm xpi && mv @idea-town-addon-0.0.1.xpi dist/idea-town-addon-0.0.1.xpi"
   },
   "license": "MPL-2.0",

--- a/stub-content/index.js
+++ b/stub-content/index.js
@@ -4,10 +4,9 @@ function statusUpdate(msg, detail) {
   var p = document.createElement('p');
   p.textContent = msg;
   statusBox.appendChild(p);
-  console.log('STATUS UPDATE:: ', msg, ' :: ', detail);
 }
 
-window.addEventListener("from-addon-to-web", function (event) {
+window.addEventListener('from-addon-to-web', function(event) {
   if (!event.detail || !event.detail.type) { return; }
   statusUpdate(event.detail.type, event.detail);
   switch (event.detail.type) {
@@ -58,7 +57,7 @@ function renderUpdates(ev) {
 document.querySelector('.update-submit').onclick = function(ev) {
   var approvedUpdates = Array.slice.call(0, document.querySelectorAll('.updates li')).filter(function(el) {
                           return (el.querySelector('input').checked);
-  }).map(function(el) {return el.textContent});
+  }).map(function(el) {return el.textContent;});
 
   sendToAddon({type: 'update-approve', detail: approvedUpdates});
-}
+};

--- a/test/test-index.js
+++ b/test/test-index.js
@@ -1,12 +1,12 @@
-var main = require("../");
+// const main = require('../');
 
-exports["test main"] = function(assert) {
-  assert.pass("Unit test running!");
+exports['test main'] = function(assert) {
+  assert.pass('Unit test running!');
 };
 
-exports["test main async"] = function(assert, done) {
-  assert.pass("async Unit test running!");
+exports['test main async'] = function(assert, done) {
+  assert.pass('async Unit test running!');
   done();
 };
 
-require("sdk/test").run(exports);
+require('sdk/test').run(exports);

